bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSNodeHandler.java

```diff
@@ -457,101 +488,121 @@ private boolean setupThing(Node node) {
 
         configurationAsChannels = Objects.requireNonNull(getBridge()).getConfiguration()
                 .as(ZwaveJSBridgeConfiguration.class).configurationChannels;
+
+        ZwaveJSTypeGeneratorResult result;
         try {
-            ZwaveJSTypeGeneratorResult result = typeGenerator.generate(thing.getUID(), node, configurationAsChannels);
+            result = typeGenerator.generate(thing.getUID(), node, configurationAsChannels);
+        } catch (Exception e) {
+            logger.error("Node {}. Error generating type information", node.nodeId, e);
+            return false;
+        }
 
-            ThingBuilder builder = editThing();
-            if (!result.location.equals(getThing().getLocation()) && !result.location.isBlank()) {
-                builder.withLocation(result.location);
-            }
+        ThingBuilder builder = editThing();
```

Comment:
You create the builder within this method, so I think you should also update the thing with bulder.build here in this same method. i.e. the other methods that add or remove parts via .withXYZ calls should not do the builder.build but should only return the modified builder. Maybe even via a try .. finally construct..

```
try {
  builder = editThing();
  builder = addSomethingTo(builder, ..);
  builder = addRemoveSomethingFrom(builder, ..);
  ..
} finally {
  updateThing(builder.build());
}
```

Reply:
The updateThing was in the right codepath, but made the two methods dependent. Changed it and now it is more intuitive.